openshift.py

```diff
@@ -0,0 +1,750 @@
+import kubernetes
+import pprint
+import logging
+import requests
+import json
+import re
+from flask import Flask, redirect, url_for, request, Response
+
+import sys
+
+# application = Flask(__name__)
+
+
+class openshift:
+    headers = None
+    verify = False
+    url = None
+
+    def __init__(self, url, token, logger):
+        self.set_token(token)
+        self.set_url(url)
+        self.logger = logger
+
+    def set_token(self, token):
+        self.headers = {
+            "Authorization": "Bearer " + token,
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+
+    def set_url(self, url):
+        self.url = url
+
+    def get_url(self):
+        return self.url
+
+    def cnvt_project_name(self, project_name):
+        suggested_project_name = re.sub("^[^A-Za-z0-9]+", "", project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9]+$", "", suggested_project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9\-]+", "-", suggested_project_name)
+        return suggested_project_name
+
+    def get_request(self, url, debug=False):
+        r = requests.get(url, headers=self.headers, verify=self.verify)
+        if debug == True:
```

Comment:
Opportunity to be more Pythonic: You can do `self.logger.debug` and if the logging level is set to debug it will log at the debug level. It seems wrong to me log a debug message at the info level. The way you have structured the messages below also aren't really helpful since they will all appear in separate lines, and since you have multiple threads, they will all be mixed up. I would suggest you having one line with all the information for the current event at once.

Reply:
The reason why I did that was to be able to turn off the debug messages from higher in the call tree.  This way I can turn off debugging messages and minimize the number of messages I need to process.